src/lib.rs

```diff
@@ -382,6 +382,8 @@ mod tests {
         )
         .unwrap();
 
-        assert_eq!(img.read_pixel_bytes()[0..5], [1, 1, 1, 255, 1]);
+        let bytes = img.save_to_bufferv("png", &[]).unwrap();
+
+        assert_eq!(bytes[1000..1005], [77, 210, 4, 80, 15]);
```

Comment:
I'm not sure whether we should rely on the PNG encoding in test cases, since encodings are not fixed. The raw image data sounds like a more useful test vector, no?

Reply:
The raw image data is unsuitable for this testcase as the `rowstride` is not used for this (i.e. we directly save the image pixels to the to the raw image data and read it out, there is no rowstride required and the bug therefore does not occure). When encoding into a PNG, GDK will use the rowstride to save it.